### ISSUE TITLE
Consider modifying return values of Leads class methods

### ISSUE THREAD

**Issue by dyba** [pennymac/velocify#2]
It's not intuitive how to use ```Velocify::ResponseReader``` with a ```Velocify::Lead``` object:

``` ruby
>> # this is a bit weird, having to dig into the hash
>> lead = Velocify::Lead.find_by_id('12345')[:leads][:lead] 
>>  # so that we can use it here like so
>> field_reader = Velocify::ResponseReader.read(kind: :field, response: lead)
>> field_reader.search_by_title 'Name'
=> [
    {:@field_id=>"2", :@value=>"Some", :@field_title=>"First Name", :@field_type=>"Text"},
    {:@field_id=>"3", :@value=>"One", :@field_title=>"Last Name", :@field_type=>"Text"},
    {:@field_id=>"670", :@value=>"Cherry Pie", :@field_title=>"Campaign Name", :@field_type=>"Text"}
]
```

We can start by documenting it in the README, but perhaps we may want to consider modifying ```Velocify::Lead.find_by_id``` to return a response that accesses :leads -> :lead for us.

**Comment by dyba** [comment]
This extends beyond the ```Leads``` class: return values for methods in model classes ought to return an array of hashes. Velocify returns a SOAP response that could be a single entry or multiple entries:

``` ruby
Velocify::Lead.find_by_phone '(323) 111-7777'
# => {:leads => :lead => { ... }}
Velocify::Lead.find_by_phone '(213) 777-1111'
# => {:leads => :lead => [{ ... }]}
```

Desired outcome:

``` ruby
Velocify::Lead.find_by_phone '(777) 123-4567'
# => [{ ... }]
# or
# => [{ ... }, ... ]
# or
# => []
```

**Comment by dyba** [comment]
Addressed in version 0.1.3
